hexapod:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 300
  
  # Position Controllers ---------------------------------------
  joint0_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: body_leg_0
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint1_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: body_leg_1
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint2_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: body_leg_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint3_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: body_leg_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint4_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: body_leg_4
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint5_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: body_leg_5
    pid: {p: 50.0, i: 0.01, d: 1.0}
#------------------------------------------------------
  joint0_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_0_1_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint1_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_1_1_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint2_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_2_1_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint3_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_3_1_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint4_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_4_1_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint5_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_5_1_2
    pid: {p: 50.0, i: 0.01, d: 1.0}

#----------------------------------------------------------
  joint0_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_0_2_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint1_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_1_2_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint2_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_2_2_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint3_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_3_2_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint4_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_4_2_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

  joint5_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: leg_5_2_3
    pid: {p: 50.0, i: 0.01, d: 1.0}

